apps/namegraph.dev/components/mini-apps/explore-collections/suggestion-category.tsx

```diff
@@ -0,0 +1,201 @@
+  top position yet, since these are separated by a white spacing,
+  we want to consider category C+1 as active. This is what the below
+  constant is used for: it is the amount of pixels that we consider between
+  the top of a category and the top of NameIdeas container to consider it active.
+*/
+const CATEGORY_TOP_OFFEST_TO_CONSIDER_IT_ACTIVE = 32;
+
+export const SuggestionCategory = ({ category }: SuggestionCategoryProps) => {
+  const [relatedCollectionsList, setRelatedCollectionsList] = useState<
+    NameGraphRelatedCollectionResponse[] | null
+  >(null);
+
+  useEffect(() => {
+    if (category) {
+      if (
+        category?.type === NameGraphGroupingCategory.related &&
+        category.related_collections
+      ) {
+        setRelatedCollectionsList(category.related_collections);
+      }
+    }
+  }, [category]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setCategoryDimensions();
+    }, 4000);
+  }, []);
+
+  const setCategoryDimensions = () => {
+    if (!category) return;
+
+    const categoryElm = document.getElementById(getCategoryID(category));
+    const categoryTop = categoryElm?.getBoundingClientRect().top;
+    const categoryBottom = categoryElm?.getBoundingClientRect().bottom;
+
+    const wrapperElm = document.getElementById("scrollable-elm");
+    const wrapperScrollTop = Number(wrapperElm?.scrollTop);
+
+    let categoryTopRelativeToWrapper =
+      Number(categoryTop) - CATEGORY_TOP_OFFEST_TO_CONSIDER_IT_ACTIVE;
+    let categoryBottomRelativeToWrapper = Number(categoryBottom);
+
+    if (wrapperScrollTop > 0) {
+      categoryTopRelativeToWrapper =
+        categoryTopRelativeToWrapper + Number(wrapperScrollTop);
+      categoryBottomRelativeToWrapper =
+        categoryBottomRelativeToWrapper + Number(wrapperScrollTop);
+    }
+
+    if (categoryElm) {
+      categoryElm.setAttribute(
+        "data-category-top",
+        categoryTopRelativeToWrapper.toString(),
+      );
+      categoryElm.setAttribute(
+        "data-category-bottom",
+        categoryBottomRelativeToWrapper.toString(),
+      );
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener(
+      "scroll",
+      lodash.debounce(setCategoryDimensions, 100),
+    );
+
+    return () => {
+      window.removeEventListener(
+        "scroll",
+        lodash.debounce(setCategoryDimensions, 100),
+      );
+    };
+  }, []);
+
+  const loadingSkeletonMarkup = (
+    <div className="w-full">
+      {category ? (
+        <>{category.name}</>
+      ) : (
+        <SuggestionCategoryHeader category={category} />
+      )}
+      <div className="mt-3">
+        <Skeleton />
+      </div>
+    </div>
+  );
+
+  useEffect(() => {
+    window.addEventListener(
+      "resize",
+      lodash.debounce(setCategoryDimensions, 100),
+    );
+
+    return () => {
+      window.removeEventListener(
+        "resize",
+        lodash.debounce(setCategoryDimensions, 100),
+      );
+    };
+  }, []);
+
+  const customizedPillsColors = [
```

Comment:
Each time I scroll down the colors are changing. This is confusing to users because they naturally then try to figure out if there's some symbolic meaning for a given color. As I understand, there's no symbolic meaning to these colors. Therefore suggest to completely remove the "dynamic rainbow" effect. If there's a goal for this please help me understand what it is. Thanks.

Reply:
I used to like how the colors would change, subjectively representing the numerous possibilities of results our NameGraph can result in. I did the requested change, though, as I understand the new perspective shared!